src/room/RTCEngine.ts

```diff
@@ -372,13 +376,20 @@ export default class RTCEngine extends EventEmitter {
 
     const startTime = (new Date()).getTime();
 
-    while ((new Date()).getTime() - startTime < maxICEConnectTimeout) {
+    let now = startTime;
+    while (now - startTime < maxICEConnectTimeout) {
+      // if there is no connectionstatechange callback fired
+      // check connectionstate after maxReconnectingTimeout
```

Comment:
nit: update comment too

```suggestion
      // check connectionstate after minReconnectWait
```